File: file-handling-io/FileHandlingIo.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Text.Json;

class FileHandlingIo
{
    static void Main()
    {
        Console.WriteLine("Language: C#");

        string baseDir = AppContext.BaseDirectory;
        string dataDir = Path.Combine(baseDir, "data");
        Directory.CreateDirectory(dataDir);

        string textPath = Path.Combine(dataDir, "sample.txt");
        string jsonPath = Path.Combine(dataDir, "sample.json");
        string csvPath = Path.Combine(dataDir, "sample.csv");

        // Buffered text write/read.
        using (var writer = new StreamWriter(textPath, false, new UTF8Encoding(false), bufferSize: 8192))
        {
            writer.WriteLine("line-1");
            writer.WriteLine("line-2");
        }

        using (var reader = new StreamReader(textPath, Encoding.UTF8, detectEncodingFromByteOrderMarks: true, bufferSize: 8192))
        {
            Console.WriteLine("Text read:");
            Console.WriteLine(reader.ReadToEnd().Trim());
        }

        // JSON write/read.
        var payload = new
        {
            name = "Nazmul",
            skills = new[] { "python", "javascript", "java", "csharp", "cpp" },
            active = true
        };

        string json = JsonSerializer.Serialize(payload, new JsonSerializerOptions { WriteIndented = true });
        File.WriteAllText(jsonPath, json);

        var parsedJson = JsonSerializer.Deserialize<Dictionary<string, JsonElement>>(File.ReadAllText(jsonPath));
        Console.WriteLine("JSON parsed keys: " + string.Join(", ", parsedJson!.Keys));

        // CSV write/read (simple parser).
        File.WriteAllLines(csvPath, new[]
        {
            "id,name,score",
            "1,Alice,95",
            "2,Bob,88"
        });

        var lines = File.ReadAllLines(csvPath);
        var headers = lines[0].Split(',');
        var parsedCsv = new List<Dictionary<string, string>>();
        for (int i = 1; i < lines.Length; i++)
        {
            var parts = lines[i].Split(',');
            var row = new Dictionary<string, string>();
            for (int j = 0; j < headers.Length; j++)
            {
                row[headers[j]] = parts[j];
            }
            parsedCsv.Add(row);
        }
        Console.WriteLine("CSV parsed rows: " + parsedCsv.Count);

        // Path API examples.
        Console.WriteLine("Path exists: " + File.Exists(textPath));
        Console.WriteLine("Path fileName: " + Path.GetFileName(textPath));
        Console.WriteLine("Path directory: " + Path.GetDirectoryName(textPath));
    }
}

File: file-handling-io/FileHandlingIo.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class FileHandlingIo {

    static String toJson(Map<String, Object> payload) {
        // Minimal JSON serializer for demonstration without external libraries.
        @SuppressWarnings("unchecked")
        List<String> skills = (List<String>) payload.get("skills");
        StringBuilder sb = new StringBuilder();
        sb.append("{\n");
        sb.append("  \"name\": \"").append(payload.get("name")).append("\",\n");
        sb.append("  \"skills\": [");
        for (int i = 0; i < skills.size(); i++) {
            sb.append("\"").append(skills.get(i)).append("\"");
            if (i + 1 < skills.size()) sb.append(", ");
        }
        sb.append("],\n");
        sb.append("  \"active\": ").append(payload.get("active")).append("\n");
        sb.append("}\n");
        return sb.toString();
    }

    static Map<String, String> parseSimpleJson(String json) {
        // Tiny parser for this known structure; keeps example dependency-free.
        Map<String, String> result = new HashMap<>();
        for (String line : json.split("\\n")) {
            String trimmed = line.trim();
            if (trimmed.startsWith("\"name\"")) {
                result.put("name", trimmed.split(":")[1].replace("\"", "").replace(",", "").trim());
            }
            if (trimmed.startsWith("\"active\"")) {
                result.put("active", trimmed.split(":")[1].replace(",", "").trim());
            }
        }
        return result;
    }

    public static void main(String[] args) throws IOException {
        System.out.println("Language: Java");

        Path baseDir = Paths.get(System.getProperty("user.dir"));
        Path dataDir = baseDir.resolve("data");
        Files.createDirectories(dataDir);

        Path textPath = dataDir.resolve("sample.txt");
        Path jsonPath = dataDir.resolve("sample.json");
        Path csvPath = dataDir.resolve("sample.csv");

        // Buffered text write/read.
        try (BufferedWriter writer = Files.newBufferedWriter(textPath, StandardCharsets.UTF_8)) {
            writer.write("line-1\n");
            writer.write("line-2\n");
        }

        try (BufferedReader reader = Files.newBufferedReader(textPath, StandardCharsets.UTF_8)) {
            System.out.println("Text read:");
            reader.lines().forEach(System.out::println);
        }

        // JSON write/read (manual minimal parser/serializer).
        Map<String, Object> payload = new HashMap<>();
        payload.put("name", "Nazmul");
        payload.put("skills", List.of("python", "javascript", "java", "csharp", "cpp"));
        payload.put("active", true);

        Files.writeString(jsonPath, toJson(payload), StandardCharsets.UTF_8);
        String jsonContent = Files.readString(jsonPath, StandardCharsets.UTF_8);
        Map<String, String> parsedJson = parseSimpleJson(jsonContent);
        System.out.println("JSON parsed (partial): " + parsedJson);

        // CSV write/read (simple parser).
        List<String> csvLines = List.of(
            "id,name,score",
            "1,Alice,95",
            "2,Bob,88"
        );
        Files.write(csvPath, csvLines, StandardCharsets.UTF_8);

        List<String> readLines = Files.readAllLines(csvPath, StandardCharsets.UTF_8);
        String[] headers = readLines.get(0).split(",");
        List<Map<String, String>> parsedCsv = new ArrayList<>();
        for (int i = 1; i < readLines.size(); i++) {
            String[] parts = readLines.get(i).split(",");
            Map<String, String> row = new HashMap<>();
            for (int j = 0; j < headers.length; j++) {
                row.put(headers[j], parts[j]);
            }
            parsedCsv.add(row);
        }
        System.out.println("CSV parsed: " + parsedCsv);

        // Path API examples.
        System.out.println("Path exists: " + Files.exists(textPath));
        System.out.println("Path fileName: " + textPath.getFileName());
        System.out.println("Path parent: " + textPath.getParent());
    }
}
